runtime/src/lib.rs

```diff
@@ -21,22 +21,33 @@ pub struct Config;
 /// The total supply of the native denomination token.
 const NATIVE_TOKEN_SUPPLY: u128 = 100_000_000 * 1_000_000_000_000_000_000;
 
+/// Determine whether the build is for Devnet.
+///
+/// If the crate version has a pre-release component (e.g. 3.0.0-alpha-devnet) then the build is
+/// classified as Devnet if that pre-release component contains "devnet".
+const fn is_devnet() -> bool {
+    const_str::contains!(env!("CARGO_PKG_VERSION_PRE"), "devnet")
+}
+
 /// Determine whether the build is for Testnet.
 ///
-/// If the crate version has a pre-release component (e.g. 3.0.0-alpha) then the build is classified
-/// as Testnet. If there is no such component (e.g. 5.0.0) then it is classified as Mainnet.
+/// If the crate version has a pre-release component (e.g. 3.0.0-alpha-testnet) then the build is
+/// classified as Testnet if that pre-release component contains "testnet".
 const fn is_testnet() -> bool {
-    !env!("CARGO_PKG_VERSION_PRE").is_empty()
+    const_str::contains!(env!("CARGO_PKG_VERSION_PRE"), "testnet")
 }
 
-/// Determine EVM chain ID to use depending on whether the build is for Localnet, Testnet or Mainnet.
+/// Determine EVM chain ID to use depending on whether the build is for Devnet, Testnet or Mainnet.
 const fn chain_id() -> u64 {
     if option_env!("OASIS_UNSAFE_USE_LOCALNET_CHAINID").is_some() {
         // Localnet.
-        0x7ec7
+        0x7ec6
+    } else if is_devnet() {
+        // Devnet.
+        0x7ec8
     } else if is_testnet() {
         // Testnet.
-        0x7ec8
+        0x7ec7
```

Comment:
I would rather do it as:
* `0x7ec7` - localnet
* `0x7ec8` - devnet
* `0x7ec9` - testnet

As this avoids changing the localnet and also doesn't result in a "strange" order where devnet comes after testnet.